app/containers/markdown/components/Image.tsx

```diff
@@ -19,17 +20,32 @@ type TMarkdownImage = {
 
 const ImageProgress = createImageProgress(ExpoImage);
 
-const MarkdownImage = ({ img, theme }: TMarkdownImage) => (
-	<ImageProgress
-		style={[styles.inlineImage, { borderColor: themes[theme].strokeLight }]}
-		source={{ uri: encodeURI(img) }}
-		indicator={Progress.Pie}
-		indicatorProps={{
-			color: themes[theme].fontHint
-		}}
-		contentFit='contain'
-	/>
-);
+const { width: screenWidth } = Dimensions.get('window');
+
+const MarkdownImage = ({ img, theme }: TMarkdownImage) => {
+	const [imageSize, setImageSize] = useState({ width: 100, height: 100 });
+
+	useEffect(() => {
+		RNImage.getSize(img, (width, height) => {
+			// Scale the image to fit within the screen width
+			const maxWidth = screenWidth * 0.9; // Max width is 90% of screen
```

Comment:
We should do something like this https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/a915a72fea1715865a86cd5acd28c93528e6a3ef/app/containers/message/Components/Attachments/Image/Image.tsx#L26